semgrep-core/src/core/ast/AST_generic.ml

```diff
@@ -602,6 +603,18 @@ and literal =
   (* Go, Python *)
   | Ratio of string wrap
 
+(*
+(*
+   A regexp is a list of template fragment. For example, /a*/ has one fragment
+   and the Ruby template /hello #{name}/ has two fragments.
+*)
+and regexp_fragment =
```

Comment:
We might also want to have a StringLiteral of string wrap bracket | RegexpLiteral of string wrap bracket * string wrap option for the easy cases, and then String of string_or_regexp_fragment list bracket | Regexp of string_or_regexp_fragment list bracket * string wrap option

Reply:
And maybe at some point we can get rid of StringLiteral and RegexpLiteral if we feel the special case is not very useful (but for constant propagation they are useful at least).

Reply:
What I did today is split regexps into literals and templates. Templates are still a list of expressions because it's simpler than having a new type for just the kind of expression a template may hold. We can always improve this later.